src/main/kotlin/ecommerce/ProductController.kt

```diff
@@ -0,0 +1,60 @@
+package ecommerce
+
+import org.springframework.http.ResponseEntity
+import org.springframework.stereotype.Controller
+import org.springframework.ui.Model
+import org.springframework.web.bind.annotation.DeleteMapping
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PatchMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.ResponseBody
+import java.net.URI
+
+@Controller
+class ProductController(private val productRepository: ProductRepository) {
+    @PostMapping("/products")
+    @ResponseBody
+    fun create(
+        @RequestBody product: Product,
+    ): ResponseEntity<Unit> {
+        val id = productRepository.insertWithKeyHolder(product)
+        return ResponseEntity.created(URI.create("/products/$id")).build()
+    }
+
+    @GetMapping("/products")
+    @ResponseBody
+    fun read(): ResponseEntity<List<Product>> {
+        val products = productRepository.findAllProducts()
+        return ResponseEntity.ok().body(products)
+    }
+
+    @PatchMapping("/products/{id}")
+    @ResponseBody
+    fun update(
+        @RequestBody newProduct: Product,
+        @PathVariable id: Long,
+    ): ResponseEntity<Unit> {
+        if (!productRepository.update(newProduct, id))
+            return ResponseEntity.notFound().build()
+        return ResponseEntity.ok().build()
+    }
+
```

Comment:
> Put creates a new product if the requested product does not exist, patch only updates existing products. Maybe patch is more accurate for our purposes? Since we have create anyways.

Hmm, is that really the difference? 🤔 
https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Methods/PATCH